src/nifgen/examples/nifgen_script.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+import argparse
+import nifgen
+import numpy as np
+from scipy import signal
+import sys
+import time
+
+
+def generate_sinewave():
+    time = np.arange(0, 10, 0.01)                 # np.arange(Start, Stop, Step)
+    amplitude = np.sin(time)
+    sinewave = amplitude.tolist()                 # List of Float
+    return sinewave
+
+
+def generate_rampup():
+    ramp = np.arange(0, 0.5, 0.01)                # numpy.arange([start,]stop, [step,]dtype=None(
+    ramp_up = ramp.tolist()                       # List of Float
+    return ramp_up
+
+
+def generate_ramdown():
+    ramp = np.arange(0, 0.5, 0.01)                # numpy.arange([start,]stop, [step,]dtype=None)
+    ramp_down = ramp.tolist()                     # List of Float
+    ramp_down.reverse()                           # Reverse list to get a ramp down
+    return ramp_down
+
+
+def generate_square():
+    time = np.arange(0, 10, 0.01)                 # np.arange(Start, Stop, Step)
+    square_build = signal.square(time, 0.5)       # signal.square(time,duty_cycle)
+    square = square_build.tolist()                # List of Float
+    return square
+
+
+def generate_triangle():
+    time = np.linspace(0, 1, 200)                 # np.arange(Start, Stop, Step)
+    triangle_build = signal.sawtooth(time)        # signal.square(time,width)
+    triangle = triangle_build.tolist()            # List of Float
+    return triangle
+
+
+def generate_gaussian_noise():
+    random_noise = np.random.normal(0, 0.1, 300)  # random.normal(center, scale, size)
+    noise = random_noise.tolist()
+    return noise
+
+
+def example(resource_name, options, shape, channel):
+    with nifgen.Session(resource_name=resource_name, options=options, channel_name=channel) as session:
+        # CONFIGURATION
+        # 1-Set the mode to Script
+        session.output_mode = nifgen.OutputMode.SCRIPT
+
+        # 2-Configure Trigger:
+        # SOFTWARE TRIGGER: used in the script
+        session.script_triggers[0].script_trigger_type = nifgen.ScriptTriggerType.SOFTWARE_EDGE  # TRIG_NONE / DIGITAL_EDGE / DIGITAL_LEVEL / SOFTWARE_EDGE
+        session.script_triggers[0].digital_edge_script_trigger_edge = nifgen.ScriptTriggerDigitalEdgeEdge.RISING  # RISING / FAILING
+        # PFI TRIGGER: not used right now
+        # session.script_triggers[1].script_trigger_type = nifgen.script_triggers[0].ScriptTriggerType.DIGITAL_EDGE
+        # session.script_triggers[1].digital_edge_script_trigger_source = 'PFI0'
+        # session.script_triggers[1].digital_edge_script_trigger_edge = session.script_triggers[0].ScriptTriggerDigitalEdgeEdge.RISING
+
+        # 3-Generate Signal
```

Comment:
Since we are in "arbitrary waveform generator" world, calling functions "generate something" is confusing, because one would naturally assume that the device will generate the signal which is not the case.

Perhaps use "calculate" as the term.